### PR TITLE
feat(Homebrew): AND-1238 Add phone number screen + tests

### PR DESCRIPTION
**Pull request by ditn**
screenshot_1534763883

### CODE REVIEW

File: kyc/src/main/java/com/blockchain/kycui/mobile/entry/KycMobileEntryPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                        // Ignore error
+                        Timber.e(it)
+                    }
+                )
+
+        compositeDisposable +=
+            view.phoneNumber
+                .doOnError(Timber::e)
+                .subscribeBy(
+                    onNext = { enableButtonIfComplete(it) },
```
Comment (westonal) [inline kyc/src/main/java/com/blockchain/kycui/mobile/entry/KycMobileEntryPresenter.kt]: You seem to check the length of the raw, unsanitized phone number here.

File: kyc/src/main/java/com/blockchain/kycui/mobile/entry/KycMobileEntryFragment.kt [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        compositeDisposable +=
+            editTextPhoneNumber
+                .onDelayedChange(KycStep.MobileNumberEntered)
+                .subscribe()
+
+        compositeDisposable +=
+            buttonNext.clicks()
+                .debounce(500, TimeUnit.MILLISECONDS, AndroidSchedulers.mainThread())
+                .doOnNext { presenter.onContinueClicked() }
+                .subscribe()
```
Comment (westonal) [inline kyc/src/main/java/com/blockchain/kycui/mobile/entry/KycMobileEntryFragment.kt]: I think you might be debouncing on main just to make sure it stays on main. Instead you should be specific about which scheduler you want to `observeOn`.

```
        compositeDisposable +=
            buttonNext.clicks()
                .debounce(500, TimeUnit.MILLISECONDS)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe { presenter.onContinueClicked() }
```

Comment (ditn) [reply kyc/src/main/java/com/blockchain/kycui/mobile/entry/KycMobileEntryFragment.kt]: Yeah the default scheduler for debouncing is `computation`, which has bitten me a few times. Specifying the `observeOn` scheduler may be better.

File: kyc/src/main/java/com/blockchain/kycui/mobile/entry/KycMobileEntryFragment.kt [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    }
+
+    override fun onPause() {
+        super.onPause()
+        compositeDisposable.clear()
+    }
+
+    override fun preFillPhoneNumber(phoneNumber: String) {
+        val formattedNumber = if (AndroidUtils.is21orHigher()) {
+            PhoneNumberUtils.formatNumber(phoneNumber, Locale.getDefault().isO3Country)
```
Comment (westonal) [inline kyc/src/main/java/com/blockchain/kycui/mobile/entry/KycMobileEntryFragment.kt]: Can you explain what users `isO3Country` does to the `formatNumber` method here?

File: kyc/src/main/java/com/blockchain/kycui/mobile/entry/KycMobileEntryFragment.kt [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    private val presenter: KycMobileEntryPresenter by inject()
+    private val progressListener: KycProgressListener by ParentActivityDelegate(this)
+    private val compositeDisposable = CompositeDisposable()
+    private val phoneNumberSubject = PublishSubject.create<String>()
+    override val phoneNumber: Observable<String> = phoneNumberSubject
+        .replayingShare()
+    private var progressDialog: MaterialProgressDialog? = null
+    private val prefixGuess by unsafeLazy {
+        "+" + PhoneNumberUtil.createInstance(context)
+            .getCountryCodeForRegion(Locale.getDefault().country)
```
Comment (westonal) [inline kyc/src/main/java/com/blockchain/kycui/mobile/entry/KycMobileEntryFragment.kt]: Use selected country for more accurate guess here?

File: kyc/src/main/java/com/blockchain/kycui/mobile/entry/KycMobileEntryFragment.kt (hunk 3 above)
Comment (ditn) [reply kyc/src/main/java/com/blockchain/kycui/mobile/entry/KycMobileEntryFragment.kt]: Kotlin has shortened it so it looks weird, it's actually `getISO3Country` and returns a country code. `formatNumber` specifically requires an ISO-3166-1 code.

**Pull request merged**